header-footer-grid/Core/Builder/Header.php

```diff
@@ -130,9 +149,181 @@ public function customize_register( WP_Customize_Manager $wp_customize ) {
+				'tab'                => 'style',
+				'type'               => 'neve_toggle_control',
+				'priority'           => 25,
+				'transport'          => 'refresh',
+				'sanitize_callback'  => 'neve_sanitize_checkbox',
+				'default'            => true,
+				'conditional_header' => true,
+			]
+		);
+
+		SettingsManager::get_instance()->add(
+			[
+				'id'                    => self::BACKGROUND_SETTING,
+				'group'                 => $section_id,
+				'section'               => $section_id,
+				'tab'                   => 'style',
+				'label'                 => esc_html__( 'Global Background', 'neve' ),
+				'description'           => esc_html__( 'A background color or image that spans across all header rows.', 'neve' ),
+				'type'                  => 'neve_background_control',
+				'live_refresh_selector' => true,
+				'live_refresh_css_prop' => [
+					'cssVar' => [
+						'vars'     => 'backgroundControl',
+						'selector' => '.global-styled',
+					],
+				],
+				'options'               => [
+					'priority'        => 30,
+					'active_callback' => [ $this, 'background_options_active_callback' ],
+				],
+				'default'               => [
+					'type'       => 'color',
+					'colorValue' => $this->default_background,
+				],
+				'transport'             => 'postMessage',
+				'sanitize_callback'     => 'neve_sanitize_background',
+				'conditional_header'    => true,
+			]
+		);
+
+		$rows = [ 'top', 'main', 'bottom' ];
+		foreach ( $rows as $row ) {
+			SettingsManager::get_instance()->add(
+				[
+					'id'                => $row . '_shortcut',
+					'group'             => $section_id,
+					'section'           => $section_id,
+					'tab'               => 'style',
+					'transport'         => 'postMessage',
+					'sanitize_callback' => 'esc_attr',
+					'type'              => '\Neve\Customizer\Controls\Button',
+					'priority'          => 35,
+					'options'           => [
+						// translators: %s row name
+						'text_before'      => sprintf( __( 'Set Background for the %s Row', 'neve' ), ucfirst( $row ) ),
+						'text_after'       => '.',
+						'button_text'      => __( 'here', 'neve' ),
+						'button_class'     => 'button_background',
+						'shortcut'         => true,
+						'is_button'        => false,
+						'control_to_focus' => 'hfg_header_layout_' . $row . '_background',
+						'active_callback'  => function () {
+							return get_theme_mod( 'neve_pro_global_header_settings_advanced_style', true );
+						},
+					],
+				]
+			);
+		}
+	}
+
+	/**
+	 * Adds a class to the header wrapper.
+	 *
+	 * @param string $classes Existing classes.
+	 * @return string Updated classes.
+	 */
+	public function add_class_to_header_wrapper( $classes ) {
+		return $classes . $this->background_options_active_callback() ? ' global-styled' : '';
+	}
+
+	/**
+	 * Returns true if individual options from header background are disabled
+	 *
+	 * @return bool
+	 */
+	public function background_options_active_callback() {
```

Comment:
We can rename this method to something like `has_global_background` as we use it in other contexts besides the callback. 🚀